qcfractal/interface/collections/collection.py

```diff
@@ -180,26 +182,76 @@ def from_json(cls, data: Dict[str, Any], client: "FractalClient" = None) -> "Col
         ret = cls(name, client=client, **data)
         return ret
 
+    @classmethod
+    def from_json(cls, jsondata: Optional[str] = None, filename: Optional[str] = None, client: "FractalClient" = None) -> "Collection":
```

Comment:
Recommend explicit `from_json(cls, *, jsondata...)` here.

Reply:
I don't quite understand the motivation; can you explain?

Reply:
Sure, since `filename` and `jsondata` are ordered, the above natively gives precedences to jsondata while in reality they are mutually exclusive.

Consider:
```python
path = Path("collection.json")
from_json(path) # Fails
from_json(path.read_text()) # Works
from_json(filename=path)) # Works
```

The `*` enforces "good" behavior `from_json(jsondata=path.read_text())`. Fundamentally diligence doesn't scale, allowing poor/lazy behaviors to leak through replicates and grows within code bases over time.

Reply:
Awesome, thanks! We're in agreement.